wpilib/src/pwm.rs

```diff
@@ -0,0 +1,233 @@
+    }
+
+    /// Temporarily disables the PWM output. The next set call will re-enable.
+    pub fn set_disabled(&mut self) -> HalResult<()> {
+        hal_call!(HAL_SetPWMDisabled(self.handle))
+    }
+
+    /// Slow down the PWM signal for old devices.
+    pub fn set_period_multiplier(&mut self, mult: PeriodMultiplier) -> HalResult<()> {
+        hal_call!(HAL_SetPWMPeriodScale(self.handle, mult as i32))
+    }
+
+    /// Honestly, I have no idea what this does and it isn't documented in wpilib.
+    pub fn set_zero_latch(&mut self) -> HalResult<()> {
+        hal_call!(HAL_LatchPWMZero(self.handle))
+    }
+
+    /// Optionally eliminate the deadband from a speed controller.
+    pub fn enable_deadband_elimination(&mut self, eliminate_deadband: bool) -> HalResult<()> {
+        hal_call!(HAL_SetPWMEliminateDeadband(
+            self.handle,
+            eliminate_deadband as i32
+        ))
+    }
+
+    /// Set the bounds on the PWM pulse widths. This sets the bounds on the PWM values for a
+    /// particular type of controller. The values determine the upper and lower speeds as well as
+    /// the deadband bracket.
+    pub fn set_bounds(
+        &mut self,
+        max: f64,
+        deadband_max: f64,
+        center: f64,
+        deadband_min: f64,
+        min: f64,
+    ) -> HalResult<()> {
+        hal_call!(HAL_SetPWMConfig(
+            self.handle,
+            max,
+            deadband_max,
+            center,
+            deadband_min,
+            min
+        ))
+    }
+
+    /// Set the bounds on the PWM values. This sets the bounds on the PWM values for a particular
+    /// each type of controller. The values determine the upper and lower speeds as well as the
+    /// deadband bracket.
+    pub fn set_raw_bounds(
+        &mut self,
+        max: i32,
+        deadband_max: i32,
+        center: i32,
+        deadband_min: i32,
+        min: i32,
+    ) -> HalResult<()> {
+        hal_call!(HAL_SetPWMConfigRaw(
+            self.handle,
+            max,
+            deadband_max,
+            center,
+            deadband_min,
+            min
+        ))
+    }
+
+    /// Get the bounds on the PWM values. This Gets the bounds on the PWM values for a particular
+    /// each type of controller. The values determine the upper and lower speeds as well as the
+    /// deadband bracket.
+    pub fn get_raw_bounds(
+        &self,
+        max: &mut i32,
+        deadband_max: &mut i32,
+        center: &mut i32,
+        deadband_min: &mut i32,
+        min: &mut i32,
+    ) -> HalResult<()> {
+        hal_call!(HAL_GetPWMConfigRaw(
+            self.handle,
+            max,
+            deadband_max,
+            center,
+            deadband_min,
+            min
+        ))
+    }
+
+    /// Get the channel of this device.
+    pub fn get_channel(&self) -> i32 {
+        self.channel
+    }
+}
+
+impl Drop for PWM {
+    fn drop(&mut self) {
+        let _ = hal_call!(HAL_SetPWMDisabled(self.handle));
```

Comment:
Why the `let _ = `, I don't think its necessary.

Reply:
I assumed it was after seeing it in other places.
https://github.com/Lytigas/first-rust-competition/blob/d36567493b39d4e06d78108948637240886a3235/wpilib/src/ds.rs#L363

Reply:
Also add [`.ok()`](https://doc.rust-lang.org/std/result/enum.Result.html#method.ok) to explicitly use the result to be clear about intent.